doc/error_handling.md

```diff
@@ -14,6 +14,10 @@ If C++ exceptions are enabled (for more info see: [Setup](setup.md)), then the
 `Napi::Error` class extends `std::exception` and enables integrated
 error-handling for C++ exceptions and JavaScript exceptions.
 
+Note, that due to limitations of the N-API, if one attempt to cast the error object thrown as a primitive, an
+wrapped object will be received instead. (With properties ```isWrapObject``` and ```errorVal``` containing the primitive value thrown)
```

Comment:
Maybe just add some additional clarification that this is when the node-addon-api Error C++ object is used within the addon.